spec/lib/signed-token-request.spec.js

```diff
@@ -0,0 +1,256 @@
+"use strict";
+
+describe("SignedTokenRequest", () => {
+    var bluebird, container, fsMock, privateSigned, publicSigned, requestAsyncMock, requestHandler, requestMock, responseMock, SignedTokenRequest, signer, StreamReadable, url;
+
+    // Just some test data that is constant.
+    const account = {
+            code: "code",
+            secret: "secret",
+            id: "accountId"
+        },
+        host = "api.opentoken.io",
+        token = "token";
+
+    beforeEach(() => {
+        container = require("../../lib/container")(host, "info");
+        bluebird = container.resolve("bluebird");
+        requestMock = require("../mock/request-mock")();
+        container.register("request", requestMock);
+        responseMock = require("../mock/response-mock")();
+        container.register("logger", require("../mock/logger-mock")());
+        fsMock = jasmine.createSpyObj("fs", [
+            "createWriteStream"
+        ]);
+        container.register("fs", fsMock);
+        requestAsyncMock = jasmine.createSpy("requestAsync");
+        container.register("requestAsync", requestAsyncMock);
+        signer = container.resolve("signer");
+        requestHandler = container.resolve("requestHandler");
+        SignedTokenRequest = container.resolve("SignedTokenRequest");
+        StreamReadable = container.resolve("StreamReadable");
+        url = `https://${host}/account/${account.id}/token`;
+        privateSigned = new SignedTokenRequest(account.id, false, account.code, account.secret);
+        publicSigned = new SignedTokenRequest(account.id, true, account.code, account.secret);
+        spyOn(signer, "sign").andCallThrough();
+        spyOn(requestHandler, "createUrl").andCallThrough();
+        spyOn(requestHandler, "checkResponse").andCallThrough();
+        spyOn(requestHandler, "buildRequestOptions").andCallThrough();
+        requestMock.get.andReturn(requestMock);
+        requestMock.post.andReturn(requestMock);
+    });
+
+
+    /**
+     * @param {string} content
+     * @return {stream~Readable}
+     */
+    function createStream(content) {
+        var stream;
+
+        stream = new StreamReadable();
+        stream.push(content);
+        stream.push(null);
+
+        return stream;
```

Comment:
This is the third time I saw this code.

Maybe you should make a Jasmine helper?